docs/usage.md

```diff
@@ -91,9 +91,126 @@ For example,
 in FreeRTOS this would be every general header file that FreeRTOS provides,
 with the addition of the `FreeRTOSConfig.h` files location at a minimum.
 
-Bristlemouth can be integrated into other build systems by...
+<!--- TODO: show how other build systems can utilize bm--->
 
-<!--- Show how other build systems can utilize bm--->
+## Wrapper Functions
+
+In order to run the Bristlemouth stack,
+some functions must be implemented by the user to hook into the created library.
+These hook functions are necessary for the following subsystems:
+
+- Device Configuration
+- Device Firmware Updates (DFU)
+- RTC Time
+- Network Driver
+
+These functions are all declared in header files within `bm_core`,
+but must be defined by the integrator in their own application.
+
+Below is an example of how this would be implemented with an arbitrary header file named `some_bm_core_header.h`:
+
+`some_bm_core_header.h`
+```
+#include <stdint.h>
+
+void some_function(char *buf, uint32_t size);
+
+```
+
+`some_integrator_defined_source_file.c`
+```
+#include "some_bm_core_header.h"
+#include <stdlib.h>
+
+
+void some_function(char *buf, uint32_t size) {
+  printf("This buffer states: %.*s\n", size, buf);
+}
+```
+
+The following explains the necessary API for aforementioned subsystems.
+
+### Configuration
+
+Keypair configuration values for the Bristlemouth stack are a paramount feature of the network stack.
```

Comment:
When I read the word "keypair", I think of public key cryptography and a pair of keys, public and private. To avoid that confusion, let's talk about the config system as a simple key-value store where the keys are strings, and values are one of the listed types. We could call one key-value pair a "config" or a "config item" or an "item in the config" or a "key-value pair". All of those ways of talking about it seem clear enough to me.

Reply:
Agreed. That sounds clearer!